src/main/java/org/craftercms/studio/impl/v2/utils/TimeUtils.java

```diff
@@ -18,64 +18,112 @@
 
 import org.craftercms.studio.api.v2.utils.function.ThrowingRunnable;
 import org.slf4j.Logger;
+import org.slf4j.event.Level;
+import org.slf4j.spi.LoggingEventBuilder;
 import org.springframework.util.function.ThrowingSupplier;
 
 public class TimeUtils {
     /**
-     * High order function to inject execution time if the logger is enabled trace.
+     * High order function to inject execution time if the given logger level is enabled.
      * Instead of this method, use {@link org.craftercms.studio.api.v2.annotation.LogExecutionTime} annotation when possible
      *
      * @param supplier     method to calculate the execution time
      * @param methodLogger logger of the method
      * @param message      a message to flag the execution method
+     * @param level        the log level
      * @param <T>          generic type of the method return value
      * @return the result of the method
      */
-    public static <T> T logExecutionTimeThrowing(ThrowingSupplier<T> supplier, Logger methodLogger, String message) throws Exception {
+    public static <T> T logExecutionTimeThrowing(ThrowingSupplier<T> supplier, Logger methodLogger, String message, Level level) throws Exception {
         long startTime = 0;
-        if (methodLogger.isTraceEnabled()) {
+        LoggingEventBuilder loggingEventBuilder = methodLogger.makeLoggingEventBuilder(level);
```

Comment:
Can we avoid this method call if we're not at the requested log level?

Reply:
Updated